andela/wg-justice-league
# #154240251 Add Fitbit support

**What does this PR do?**

In this PR we added on Wger the ability to pull in weight from Fitbit. A user with accounts on both Fitbit and Wger will be able to load his weight logs on Wger.

**Description of Task to be completed?**

In this task we were required to create a Fitbit app and use our secret and client ID to allow our users to connect to Fitbit. We generate a URL that is used to redirect to Fitbit where a user allows to connect to his account and that way we can retrieve the user weight.

**How should this be manually tested?**

- Ensure you are loggen in at Wger and you have an activated Fitbit account.
- Browse to the navigation bar and under `Weight` choose `Sync Fitbit Weight `
- By clicking to the button `Grant WGER Access To Your Fitbit Data` and then `Allow`
- Now you can browse to `Weight Overview` and _ta-da_ 😀, your weight is there.

**What are the relevant pivotal tracker stories?**

#154240251 Add fitbit support

**Screenshots**

- Grant Wger access to Fitbit 

<img width="988" alt="screen shot 2018-01-26 at 11 30 39" src="https://user-images.githubusercontent.com/19430799/35431185-8dd3d64a-028c-11e8-8b5c-18b1ff4421f5.png">

- Specify the kind of data Wger will be allowed to access

<img width="1108" alt="screen shot 2018-01-26 at 11 30 56" src="https://user-images.githubusercontent.com/19430799/35431187-90b00276-028c-11e8-979f-83977b2a6635.png">

- An example of a sync by Wger to Fitbit

<img width="798" alt="screen shot 2018-01-26 at 11 33 13" src="https://user-images.githubusercontent.com/19430799/35431236-bd8ba52a-028c-11e8-9bed-1c436999ae3c.png">
